lib/src/value/calculation.dart

```diff
@@ -120,6 +122,228 @@ class SassCalculation extends Value {
     return SassCalculation._("max", args);
   }
 
+  /// Creates a `hypot()` calculation with the given [arguments].
+  ///
+  /// Each argument must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation]. It must be passed at least one argument.
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  static Value hypot(Iterable<Object> arguments) {
+    var args = _simplifyArguments(arguments);
+    if (args.isEmpty) {
+      throw ArgumentError("hypot() must have at least one argument.");
+    }
+    _verifyCompatibleNumbers(args);
+
+    var subtotal = 0.0;
+    var iterator = 0;
+    var first = arguments.elementAt(0);
+    for (var number in arguments) {
+      if (number is! SassNumber || number.hasUnit('%')) {
+        return SassCalculation._("hypot", args);
+      }
+      if (first is SassNumber) {
+        var value = number.convertValueToMatch(
+            first, "numbers[${iterator + 1}]", "numbers[1]");
+        subtotal += (number_lib.pow(SassNumber(value), SassNumber(2))).value;
```

Comment:
Given you're working with unitless numbers here, it doesn't make sense to wrap them in `SassNumbers` just to use the number_lib functions. Instead, for here and sqrt below, just use `math.pow` and `math.sqrt` (for here, you can even just do `value * value`)